### PR TITLE
[Test] Have FileCheck dump its input on failure.

### PR DESCRIPTION
**Pull request by mikeash**

### CODE REVIEW

File: test/lit.cfg [hunk 1]
```diff
@@ -1536,7 +1536,7 @@ config.substitutions.append(('%llvm-profdata', config.llvm_profdata))
 config.substitutions.append(('%llvm-cov', config.llvm_cov))
 
 config.substitutions.append(('%FileCheck',
-                             '%r %r --sanitize BUILD_DIR=%r --sanitize SOURCE_DIR=%r --use-filecheck %r %s' % (
+                             '%r %r --sanitize BUILD_DIR=%r --sanitize SOURCE_DIR=%r --use-filecheck %r --dump-input fail %s' % (
```
Comment (gottesmm) [inline test/lit.cfg]: Does the path sanitizing FileCheck let through the argument and transform it to have an '='?

**Comment by gottesmm** [comment]
But beyond that LGTM. Just make sure that is true.

File: test/lit.cfg (hunk 1 above)
Comment (mikeash) [reply test/lit.cfg]: It lets the argument through. I don't think it transforms it. The docs seem to suggest that it's acceptable with or without the `=`: https://llvm.org/docs/CommandGuide/FileCheck.html. At any rate, it does work.

**Review by gottesmm — APPROVED** [review]
LGTM

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Pull request closed without merging**